File: backend/test_elevenlabs.js
import axios from "axios";
import FormData from "form-data";
import fs from "fs";
import dotenv from "dotenv";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.join(__dirname, ".env") });

const API_KEY = process.env.ELEVENLABS_API_KEY;
const FILE_PATH = path.join(
  __dirname,
  "uploads/b170a5a41c2ff3b036d45571c9ae5dae"
); // The 22KB file

async function testElevenLabs() {
  console.log("Testing ElevenLabs API...");
  console.log(`API Key present: ${!!API_KEY}`);

  if (!fs.existsSync(FILE_PATH)) {
    console.error("Test file not found:", FILE_PATH);
    return;
  }

  const formData = new FormData();
  formData.append("file", fs.createReadStream(FILE_PATH));
  formData.append("model_id", "scribe_v1"); // Try v1 first as it might be more stable for free tier?

  try {
    console.log("Sending request...");
    const response = await axios.post(
      "https://api.elevenlabs.io/v1/speech-to-text",
      formData,
      {
        headers: {
          ...formData.getHeaders(),
          "xi-api-key": API_KEY,
        },
      }
    );
    console.log("✅ Success!");
    console.log(response.data);
  } catch (error) {
    console.error("❌ Failed:", error.message);
    if (error.response) {
      console.error("Status:", error.response.status);
      console.error("Data:", JSON.stringify(error.response.data, null, 2));
    }
  }
}

testElevenLabs();

File: backend/test_elevenlabs.py
import os
from dotenv import load_dotenv
from elevenlabs.client import ElevenLabs

# Load environment variables
load_dotenv()

# Configuration
API_KEY = os.getenv("ELEVENLABS_API_KEY")
FILE_PATH = "uploads/b170a5a41c2ff3b036d45571c9ae5dae" # The ~22KB file

def test_transcription():
    print("Testing ElevenLabs Transcription (Python)...")
    
    if not API_KEY:
        print("❌ Error: ELEVENLABS_API_KEY not found in .env")
        return

    print(f"API Key found: {API_KEY[:5]}...")
    
    if not os.path.exists(FILE_PATH):
        print(f"❌ Error: Test file not found at {FILE_PATH}")
        return

    client = ElevenLabs(api_key=API_KEY)

    try:
        print("Sending request to ElevenLabs Scribe v1...")
        
        # Open file in binary read mode
        with open(FILE_PATH, "rb") as audio_file:
            transcript = client.speech_to_text.convert(
                file=audio_file,
                model_id="scribe_v1"
            )
            
        print("\n✅ Success! Transcription result:")
        print("Language:", transcript.language_code)
        print("Text:", transcript.text)
        
    except Exception as e:
        print(f"\n❌ API Error: {str(e)}")
        # Check if it has a body/message attribute commonly returned by the SDK
        if hasattr(e, 'body'):
            print("Details:", e.body)

if __name__ == "__main__":
    test_transcription()
